redisinsight/api/src/modules/workbench/providers/command-execution.provider.ts

```diff
@@ -0,0 +1,175 @@
+import { Injectable, Logger, NotFoundException } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Repository } from 'typeorm';
+import { isUndefined, filter, isNull } from 'lodash';
+import { plainToClass } from 'class-transformer';
+import { EncryptionService } from 'src/modules/core/encryption/encryption.service';
+import { CommandExecutionEntity } from 'src/modules/workbench/entities/command-execution.entity';
+import { CommandExecution } from 'src/modules/workbench/models/command-execution';
+import ERROR_MESSAGES from 'src/constants/error-messages';
+import { classToClass } from 'src/utils';
+
+@Injectable()
+export class CommandExecutionProvider {
+  private logger = new Logger('CommandExecutionProvider');
+
+  constructor(
+    @InjectRepository(CommandExecutionEntity)
+    private readonly commandExecutionRepository: Repository<CommandExecutionEntity>,
+    private readonly encryptionService: EncryptionService,
+  ) {}
+
+  /**
+   * Encrypt command execution and save entire entity
+   * Should always throw and error in case when unable to encrypt for some reason
+   * @param commandExecution
+   */
+  async create(commandExecution: Partial<CommandExecution>): Promise<CommandExecution> {
+    const entity = plainToClass(CommandExecutionEntity, commandExecution);
+
+    return classToClass(
+      CommandExecution,
+      await this.decryptEntity(
+        await this.commandExecutionRepository.save(await this.encryptEntity(entity)),
+        true,
+      ),
+    );
+  }
+
+  /**
+   * @param databaseId
+   */
+  async getList(databaseId: string): Promise<CommandExecution[]> {
```

Comment:
It would be nice to add a description of this method, as well as done for others.

Reply:
added